pages/games/[hash].js

```diff
@@ -18,28 +15,68 @@ function Game({ data, playerHash }) {
     );
     const [gameStatus, setGameStatus] = useState(data.game_status);
     const [winner, setWinner] = useState(data.winner);
+    const [opponentHits, setOpponentHits] = useState(data.hits);
+    const [opponentShoots, setOpponentShoots] = useState(data.hits);
+    const [displayMessage, setDisplayMessage] = useState(false);
+    const [message, setMessage] = useState('');
+    const [isRefreshing, setIsRefreshing] = useState(false);
+    const [isLinkCopied, setIsLinkeCopied] = useState(false);
 
     const isCurrentPlayer = currentPlayer.hash === playerHash;
     const isGameOpen = gameStatus === GAME_OPEN;
     const isGameOver = gameStatus === GAME_OVER;
 
-    function toggleBoard() {
-        setVisibleBoard(visibleBoard === PLAYER ? OPPONENT : PLAYER);
-    }
-
     async function refreshGameState() {
+        setIsRefreshing(true);
         const res = await api.getGame(data.game, playerHash);
-        setCurrentPlayer(res.current_player);
-        setGameStatus(res.game_status);
-        setHits(res.hits);
-        setShoots(res.shoots);
-        setOpponentSunkShips(res.opponent_sunk_ships);
-        setWinner(res.winner);
+
+        setOpponentHits(res.opponent_hits);
+        setOpponentShoots(res.opponent_shoots);
+
+        setTimeout(function () {
+            setCurrentPlayer(res.current_player);
+            setGameStatus(res.game_status);
+            setHits(res.hits);
+            setShoots(res.shoots);
+            setOpponentSunkShips(res.opponent_sunk_ships);
+            setWinner(res.winner);
+            setIsRefreshing(false);
+        }, 2000);
     }
 
     async function shoot(x, y) {
+        if (isRefreshing) {
+            return;
+        }
+
         const res = await api.shoot(data.game, playerHash, [x, y]);
-        refreshGameState();
+        setHits(res.hits);
+        setShoots(res.shoots);
+
+        if (res.did_hit) {
+            setMessage('TOUCHE');
+        } else {
+            setMessage('LOUPE');
+        }
+        setDisplayMessage(true);
+
+        setTimeout(function () {
+            refreshGameState();
+            setDisplayMessage(false);
+        }, 2000);
+    }
+
+    function copyToClipBoard() {
```

Comment:
You can create a custom hook named `useCopyToClipBoard` which is responsible for copying a text and managing the timeout.